lib/connectors_async/job.rb

```diff
@@ -0,0 +1,88 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License;
+# you may not use this file except in compliance with the Elastic License.
+#
+
+require 'connectors_shared/job_status'
+
+# The class is actually supported for single-threaded usage EXCEPT for :documents field
+# :documents are a Queue that's stated to be safe in a threaded environment
+module ConnectorsAsync
+  class Job
+    class StatusUpdateError < StandardError; end
+    def initialize(job_id)
+      @data = {
```

Comment:
Should store `content_source_id`, `job_id`, `job_type` as well?

Reply:
`job_id` is stored, `job_type` we cannot determine - it's not sent by Enterprise Search and is actually not specific to the job.

As for `content_source_id` - we can, but there was never a need to do so, so I decided to not store it per job until we have a use-case for it.

Reply:
`job_type` and `content_source_id` can be sent by `ent-search`, and can be used for reject job creation, as I mentioned above.

Reply:
Indeed, but since it's not possible to interrupt a running job in the `connectors` repo, it's not possible to enforce this logic without breaking the flow heavily - re-triggering jobs in Enterprise Search. I will think of a way to approach it, but probably there's no good way without major Enterprise Search refactoring (there's no way for now even to send a cancel signal to connector package